lib/llm/LLMCache.ts

```diff
@@ -0,0 +1,139 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as crypto from "crypto";
+
+interface CacheEntry {
+  timestamp: number;
+  data: any;
+}
+
+interface CacheStore {
+  [key: string]: CacheEntry;
+}
+
+export class LLMCache {
+  private cacheDir: string;
+  private cacheFile: string;
+  private logger: (message: {
+    category?: string;
+    message: string;
+    level?: number;
+  }) => void;
+
+  private readonly CACHE_MAX_AGE_MS = 7 * 24 * 60 * 60 * 1000; // 1 week in milliseconds
+  private readonly CLEANUP_PROBABILITY = 0.01; // 1% chance
+
+  constructor(
+    logger: (message: {
+      category?: string;
+      message: string;
+      level?: number;
+    }) => void,
+    cacheDir: string = path.join(process.cwd(), "tmp", ".cache"),
+    cacheFile: string = "llm_calls.json",
+  ) {
+    this.logger = logger;
+    this.cacheDir = cacheDir;
+    this.cacheFile = path.join(cacheDir, cacheFile);
+    this.ensureCacheDirectory();
+  }
+
+  private ensureCacheDirectory(): void {
+    if (!fs.existsSync(this.cacheDir)) {
+      fs.mkdirSync(this.cacheDir, { recursive: true });
+    }
+  }
+
+  private createHash(data: any): string {
+    const hash = crypto.createHash("sha256");
+    return hash.update(JSON.stringify(data)).digest("hex");
+  }
+
+  private readCache(): CacheStore {
+    if (fs.existsSync(this.cacheFile)) {
```

Comment:
I see that we read/write the JSON every time we do an LLM call. what if we have multiple parallel processes? Should we not have the object in memory instead and instantiate higher in the abstraction?

Reply:
we might want to consider Redis down the line i guess- could open as an issue